src/blog/fearless-structured-concurrency.vmd

```diff
@@ -0,0 +1,542 @@
+---
+title: Combining Fearless Concurrency with Structured Concurrency
+author: Evan Ovadia
+date: Jan 8, 2022
+realm: blog
+path: blog/fearless-structured-concurrency
+layout: annotated
+namespace: c-blog m-annotated
+---
+
+With fearless structured concurrency, we can add multi-threading to existing code with a single keyword, with no risk of data races!
+
+
+Summary:
+
+ * "Structured" concurrency makes multi-threading much *easier*.
+    * Threads can access data from the containing scope!
+ * "Fearless" concurrency *prevents data races* between threads.
+ * *Structured fearless concurrency combines the best of both worlds.*
+ * Vale's `parallel` keyword makes it happen!
+
+
+# We Want Structured Concurrency!
+
+
+Structured concurrency is a lot of fun.
+
+
+With structured concurrency, you can launch multiple threads at once, to run your calculations in parallel, for a big speedup. And often, it can be done with a tiny adjustment.
+
+
+Imagine we had this simple program that calculated x^3 for some values of x:
+
+```c
+#include <math.h>
+#include <stdio.h>
+
+void main() {
+  int exponent = 3;
+  int results[5];
```

Comment:
I'd add a comment explaining that in practice much larger data is used, and the small array size is just an example

Reply:
Done!